### DISCUSSION TITLE
Keys missing from postscript

### DISCUSSION THREAD

**Discussion by peterstory** [retorquere/zotero-better-bibtex#2912] [Q&A]
I've been using the following postscript to add URLs to my BibTeX export:
```
if (Translator.BetterBibTeX) {
  if ((zotero.itemType == 'presentation') && (tex.referencetype == 'misc') && zotero.meetingName) {
    tex.referencetype = 'article'
    tex.add({name: 'journal', 'value': zotero.meetingName})
  }

  if ((zotero.itemType == 'webpage') && (zotero.url)) {
    tex.remove('howpublished')
    tex.add({name: 'note', bibtex: "{\\url{" + tex.enc_latex({value: zotero.url}) + "}}"})
  }
}
```

When viewing the diff of my latest export, I noticed that many entries are now missing their URLs. For example, this entry:
```diff
 @misc{advertisingstandardsauthorityASARulingTefincom2019,
   title = {{{ASA Ruling}} on {{Tefincom SA}} t/a {{NordVPN}}},
   author = {{Advertising Standards Authority}},
   year = {2019},
   month = may,
   urldate = {2021-06-30},
-  file = {/Users/pstory/Zotero/storage/W6MNGJXH/tefincom-sa-a19-547668.html},
-  note = {\url{https://www.asa.org.uk/rulings/tefincom-sa-a19-547668.html}}
+  file = {/Users/pstory/Zotero/storage/W6MNGJXH/tefincom-sa-a19-547668.html}
 }
```

I know that my postscript is running, because if I comment out `tex.remove('howpublished')` then `howpublished` appears in the export. So I don't understand why the `tex.add()` would have no effect on the export. 

A screenshot of the associated info in Zotero:
<img width="335" alt="image" src="https://github.com/retorquere/zotero-better-bibtex/assets/1322135/80e39962-97d5-411a-a6e9-e27b810080b6">

And of my export settings:
<img width="716" alt="Screenshot 2024-06-28 at 9 29 57 AM" src="https://github.com/retorquere/zotero-better-bibtex/assets/1322135/a7d06ac3-250b-4e38-a763-d26d42e88e0a">

**Comment by retorquere** [comment]
Please open an issue with a debug log.

**Reply by peterstory** [reply]
Thanks for the tip about the debug log. I've opened an issue here: https://github.com/retorquere/zotero-better-bibtex/issues/2913
